src/material-experimental/mdc-slide-toggle/_slide-toggle-theme.scss

```diff
@@ -1,103 +1,140 @@
+    selected-hover-handle-color: $state-content,
+    selected-pressed-handle-color: $state-content,
+
+    selected-focus-track-color: $inverse,
+    selected-hover-track-color: $inverse,
+    selected-pressed-track-color: $inverse,
+    selected-track-color: $inverse,
+  );
+}
 
 @mixin color($config-or-theme) {
   $config: theming.get-color-config($config-or-theme);
   $primary: theming.get-color-from-palette(map.get($config, primary));
   $accent: theming.get-color-from-palette(map.get($config, accent));
   $warn: theming.get-color-from-palette(map.get($config, warn));
-
-  // Save original values of MDC global variables. We need to save these so we can restore the
-  // variables to their original values and prevent unintended side effects from using this mixin.
-  $orig-baseline-theme-color: mdc-switch.$baseline-theme-color;
-  $orig-toggled-off-thumb-color: mdc-switch.$toggled-off-thumb-color;
-  $orig-toggled-off-track-color: mdc-switch.$toggled-off-track-color;
-  $orig-disabled-thumb-color: mdc-switch.$disabled-thumb-color;
-  $orig-disabled-track-color: mdc-switch.$disabled-track-color;
+  $is-dark: map.get($config, is-dark);
 
   @include mdc-helpers.mat-using-mdc-theme($config) {
-    mdc-switch.$baseline-theme-color: primary;
-    mdc-switch.$toggled-off-thumb-color: mdc-theme-color.prop-value(surface);
-    mdc-switch.$toggled-off-track-color: mdc-theme-color.prop-value(on-surface);
-    mdc-switch.$disabled-thumb-color: mdc-theme-color.prop-value(surface);
-    mdc-switch.$disabled-track-color: mdc-theme-color.prop-value(on-surface);
-
     // MDC's switch doesn't support a `color` property. We add support
     // for it by adding a CSS class for accent and warn style.
     .mat-mdc-slide-toggle {
       @include mdc-form-field.core-styles($query: mdc-helpers.$mat-theme-styles-query);
-
-      .mdc-switch__thumb-underlay::after, .mat-ripple-element {
-        background: mdc-switch.$toggled-off-ripple-color;
-      }
+      @include mdc-switch-theme.theme(_get-theme-base-map($is-dark));
 
       &.mat-primary {
-        @include mdc-switch.without-ripple($query: mdc-helpers.$mat-theme-styles-query);
+        @include mdc-switch-theme.theme(_get-theme-color-map($primary));
       }
 
       &.mat-accent {
-        mdc-switch.$baseline-theme-color: secondary;
-        @include mdc-switch.without-ripple($query: mdc-helpers.$mat-theme-styles-query);
+        @include mdc-switch-theme.theme(_get-theme-color-map($accent));
       }
 
       &.mat-warn {
-        mdc-switch.$baseline-theme-color: error;
-        @include mdc-switch.without-ripple($query: mdc-helpers.$mat-theme-styles-query);
+        @include mdc-switch-theme.theme(_get-theme-color-map($warn));
       }
     }
 
     // The ripple color matches the palette only when it's checked.
     .mat-mdc-slide-toggle-checked {
-      .mdc-switch__thumb-underlay::after, .mat-ripple-element {
+      .mdc-switch__ripple::after, .mat-ripple-element {
```

Comment:
I don't believe you need to override `.mdc-switch__ripple::after` for these. They should be set with `_get-theme-color-map()` by the `selected-*-state-layer-color` token